xray/segment.go

```diff
@@ -269,51 +268,81 @@ func (subseg *Segment) CloseAndStream(err error) {
 	}
 
 	subseg.beforeEmitSubsegment(subseg.parent)
-	subseg.Unlock()
-
 	subseg.emit()
 }
 
 // RemoveSubsegment removes a subsegment child from a segment or subsegment.
 func (seg *Segment) RemoveSubsegment(remove *Segment) bool {
 	seg.Lock()
-	defer seg.Unlock()
```

Comment:
Why we remove defer here? defer will be called before return.

Reply:
For fine granular locking - decrementing `totalSubSegments` count is thread safe now and hence prior to that we unlock the subsegment : [reference](https://github.com/aws/aws-xray-sdk-go/pull/140/files#diff-4178d11b198e2b2e67f13586d5f83fc5R288)